eliot-service/eliot/symbolicate_resource.py

```diff
@@ -320,49 +322,96 @@ def get_symcache(self, debug_filename, debug_id, debug_stats):
         end_time = time.perf_counter()
         debug_stats.incr("cache_lookups.time", end_time - start_time)
 
-        # We didn't find it in the cache, so try to download it
-        if symcache is None:
-            start_time = time.perf_counter()
-
-            # Download the SYM file from one of the sources
-            sym_file = self.download_sym_file(debug_filename, debug_id)
-            if sym_file is not None:
-                # Extract the module filename--this is either debug_filename or
-                # pe_filename on Windows
-                module_filename = get_module_filename(sym_file, debug_filename)
-
-                # Parse the SYM file into a symcache
-                symcache = self.parse_sym_file(debug_filename, debug_id, sym_file)
-
-                # If we have a valid symcache file, cache it to disk
-                if symcache is not None:
-                    data = symcache_to_bytes(symcache)
-
-                    data = {"symcache": data, "filename": module_filename}
-                    self.cache.set(cache_key, data)
-
-                    end_time = time.perf_counter()
-                    debug_stats.incr("downloads.count", 1)
-                    debug_stats.incr(
-                        [
-                            "downloads",
-                            "size_per_module",
-                            f"{debug_filename}/{debug_id}",
-                        ],
-                        len(sym_file),
-                    )
-                    debug_stats.incr(
-                        [
-                            "downloads",
-                            "time_per_module",
-                            f"{debug_filename}/{debug_id}",
-                        ],
-                        end_time - start_time,
-                    )
-
         if symcache:
             return symcache, module_filename
 
+        # We didn't find it in the cache, so try to download it from one of the sources
+        download_start_time = time.perf_counter()
+        sym_file = self.download_sym_file(debug_filename, debug_id)
+        download_end_time = time.perf_counter()
+
+        debug_stats.incr("downloads.count", 1)
+
+        if sym_file is None:
+            # Nothing was downloaded, so capture the timing as a download fail and
+            # return None
+            debug_stats.incr(
+                [
+                    "downloads",
+                    "fail_time_per_module",
```

Comment:
I waffled about how to represent this. We want to know timings for success _and_ failure--maybe the failure case took a really long time or something like that. I wanted to keep things in sections by action. I ended up with this.

These are debug stats, so we can always change them later.